src/snowflake/cli/api/utils/graph.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+
+
+class Graph:
+    @dataclass
+    class Node:
+        key: str
+        status: str | None = None
+        dependencies: set[Graph.Node] = field(default_factory=set)
```

Comment:
nit: edges? I think that's the standard term. Dependencies is your use for this graph.